src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.TypeNames.cs

```diff
@@ -25,10 +24,33 @@ internal string GetTypeName(TypeAndCustomInfo typeAndInfo, bool escapeKeywordIde
             var dynamicFlags = DynamicFlagsCustomTypeInfo.Create(typeAndInfo.Info);
             var index = 0;
             var pooled = PooledStringBuilder.GetInstance();
-            AppendQualifiedTypeName(pooled.Builder, type, dynamicFlags, ref index, escapeKeywordIdentifiers);
+            AppendQualifiedTypeName(pooled.Builder, type, dynamicFlags, ref index, escapeKeywordIdentifiers, out sawInvalidIdentifier);
             return pooled.ToStringAndFree();
         }
 
+        internal bool IsValidIdentifier(string name)
+        {
```

Comment:
Perhaps this should be an abstract method so derived types can use CharacterInfo.IsValidIdentifier(name). Then IsIdentifierPartCharacter could be removed as well.